packages/datatrak-web/src/features/Survey/SurveySideMenu/SurveySideMenu.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import styled from 'styled-components';
+import { Drawer as BaseDrawer, ListItem, List, ButtonProps } from '@material-ui/core';
+import { useSurveyForm } from '../SurveyContext';
+import { SideMenuButton } from './SideMenuButton';
+import { ButtonLink } from '../../../components';
+import { To } from 'react-router';
+
+export const SIDE_MENU_WIDTH = '20rem';
+
+const Drawer = styled(BaseDrawer).attrs({
+  anchor: 'left',
+  variant: 'persistent',
+  elevation: 0,
+})`
+  flex-shrink: 0;
+  width: ${SIDE_MENU_WIDTH};
+  position: relative;
+  height: 100%;
+  .MuiPaper-root {
+    width: 100%;
+    position: absolute;
+    background-color: transparent;
+    border-right: none;
+    height: 100%;
+  }
+`;
+
+const SurveyMenuContent = styled(List)`
+  padding: 0 0.5rem;
+`;
+
+const SurveyMenuItem = styled(ListItem).attrs({
+  button: true,
+  component: ButtonLink,
+  variant: 'text',
+  color: 'default',
+})<
+  ButtonProps & {
+    to: To;
+    $active?: boolean;
+  }
+>`
+  padding: 0.5rem;
+  align-items: flex-start;
+  justify-content: flex-start;
+  background-color: ${({ theme, $active }) =>
+    $active ? `${theme.palette.primary.main}55` : 'transparent'};
+  & ~ & {
+    margin-left: 0; // overwrite styles from elsewhere
+  }
+  &:hover {
+    background-color: ${({ theme }) => `${theme.palette.grey[400]}4d`};
+  }
+`;
+
+const SurveyScreenNumber = styled.span`
+  width: 2rem;
+`;
+
+const SurveyScreenTitle = styled.span`
+  width: 100%;
+  font-weight: ${({ theme }) => theme.typography.fontWeightRegular};
+`;
+
+export const SurveySideMenu = () => {
+  const { sideMenuOpen, toggleSideMenu, surveyScreenComponents, screenNumber } = useSurveyForm();
```

Comment:
I don't think the side menu open state needs to be in the context since the state could just be placed here and shared between the Drawer and  SideMenuButton. Would you mind taking it out of the context?

Reply:
I put it in the context because the screen component needs to know the side menu open status so that it can adjust the margin etc accordingly and shift to the side